actnow/petitions/models.py

```diff
@@ -21,6 +21,11 @@ class Petition(TimestampedModelMixin):
         on_delete=models.SET_NULL,
         null=True,
     )
+    followers = models.ManyToManyField(
+        ActNowUser,
+        verbose_name=_("followers"),
+        related_name="petitions",
```

Comment:
Not sure about this: If we had petitions on user model, it should be for petitions that that user has created. Petitions the user is following should be on different field.

But thinking out loud, should this be on user model at all? Up for discussion but I'm thinking it shouldn't be.

Reply:
>Petitions the user is following should be on different field.

True @kilemensi 
That's why I added the `related_name="petitions",` now we have `user.petitions` and `user.petition_set` which shows petitions belonging to a user.

>But thinking out loud, should this be on user model at all? Up for discussion but I'm thinking it shouldn't be.

Other options would be, store the followers in an `ArrayField` or use `UserProfile` instead of User model. What do you think.

Reply:
No, if they're to be accessible, then `User` model is the correct model. The question is **_should_** they be accessible from `User` model?

I'm thinking (again, out loud),

1. `/petitions/?owner=user` is best way to retrieve all petitions created by `user`, and
2. `/petitions/?followers=user` is best way to retrieve all petitions followed by `user`.

Our user model remains small and nimble.

What are the advantages of having `petitions`/`following` and` petition_set`/`petitions` on `User` model?

Reply:
I see.
To clarify,  by `petition_set` and `petitions` I meant how they are accessed from the manager. 
i.e `user.petition_set.all()` will give us all petitions belonging to this user which we can return in the `/petitions/?owner=user` endpoint. And `user.petitions.all()` will return all petitions this `user` is following which can be returned in the suggested `/petitions/?followers=user` endpoint.

Reply:
Yeah, I get you. But the question remains do we want to use `User` object to get to petitions or use `Petition` model to filter those belonging to a user or where user is one of followers?

Imagine if you can start following profile of other users that have same interests as you? We'll also need to add these relations to user account as well?

But either way, it's a small thing so if you're cool with using User model, then lets proceed with that. But that `petition_set` name needs to change.

Reply:
@kilemensi 
> But the question remains do we want to use User object to get to petitions or use Petition model to filter those belonging to a user or where user is one of followers?

This too can be achieved with the current implementation. Something like `Petition.objects.filter(owner=x)` 
Also, when we have a petition we can do `petition.followers` to return all followers.

Reply:
Using `DjangoFilterBackend` [here](https://github.com/CodeForAfrica/actNOW/pull/76/files#diff-9460dbfcd919b543e9efcdc276c52cfbc0b8baa644b91f76a5bf11b57da8b075R20) removes the need to do the manual queries.